### PR TITLE
Enable consent setting and pass setting to iframe

### PR DESCRIPTION
**Pull request by zhouyx**
Configuration looks like 
```
{
  'storageKey': ...
  'checkConsentHref': ...
  'setting': ...
  'promptUI': ...
  'postPromptUI': ...
}
```

Old config
```
{
   'storageKey': {
     'checkConsentHref': 
     'promptUI':
   }
   'setting': ...
   'postPromptUI': ...
}
   
```

The newly introduced setting object will be appended to checkConsentHref request body, and the created iframe name attribute. 

The new config format is not supported yet, this PR only converts existing format to the new format. I'll have a following PR to support new format. And please give advice on naming : )

### CODE REVIEW

**Comment by zhouyx** [comment]
This is ready for review. PTAL

File: extensions/amp-consent/0.1/consent-ui.js [hunk 1]
```diff
@@ -286,6 +291,13 @@ export class ConsentUI {
    * @return {!Promise}
    */
   loadIframe_() {
+    const info = dict({});
+    if (this.inlineSetting_) {
+      info['setting'] = this.inlineSetting_;
+    }
+
+    this.ui_.setAttribute('name', JSON.stringify(info));
```
Comment (torch2424) [inline extensions/amp-consent/0.1/consent-ui.js]: If I remember correctly, you will need to pass a sentinel (as this is what we do to communicate with 3p frames), and stringify using a `dict()`. 

See: https://github.com/ampproject/amphtml/blob/master/extensions/amp-recaptcha-input/0.1/amp-recaptcha-service.js#L222

Comment (torch2424) [reply extensions/amp-consent/0.1/consent-ui.js]: Discussed offline, don't need a sentinel for this use case, and dict was used three lines above 🙃

**Comment by zhouyx** [comment]
Talked offline. `setting` in config sounds too general. Decided to change the field name to `clientConfig`. Thank to you @torch2424 for the naming idea! Please take another look.

**Pull request merged**